src/ddscxx/src/org/eclipse/cyclonedds/domain/qos/DomainParticipantQosDelegate.cpp

```diff
@@ -103,8 +103,7 @@ DomainParticipantQosDelegate::check() const
 bool
 DomainParticipantQosDelegate::operator ==(const DomainParticipantQosDelegate& other) const
 {
-    return other.present_             == present_ &&
```

Comment:
This change in the behaviour of `operator==` (and the same for the other QoS types) I think warrants its own commit with its own justification for no longer checking whether the `present` flags are the same.

I do suspect there is a difference before-and-after, because I suspect that:
```
 f(const DataReaderQos& qa, const DataReaderQos& qb) {
  Topic<T> t(..., Reliability::Reliable());
  DataReader<T> a(..., qa);
  DataReader<T> b(..., qb);
  return a.qos() == b.qos();
}
```
will show a change in behaviour if, e.g., `qa` has `Reliability` set to `BestEffort` (the default) and `qb` has not set reliability, because in that case `b` will inherit the reliability setting from the topic, which has it set to `Reliable` (not the default).

It may be that it is currently impossible to make a sparse QoS object in C++ (I just don't know the C++ binding well enough to be sure even if I do try to check). In that case, there is no observable change in behaviour given the available interfaces.

Otherwise, I think that instead of changing `operator==` there should be an operation to complete the QoS with the default values, so that you could do `qa.dense() == qb.dense()` vs `qa == qb`. (Not that I am certain that `dense` is the best name, it is just the first thing that came to mind because I think of the QoS objects as "sparse" things.)

Then, whatever prompted this change (I suspect it is test code) could take the detour via this `dense()` function.